src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiController.java

```diff
@@ -54,6 +68,41 @@ public UsersApiController(NativeWebRequest request, UserService userService) {
         return new ResponseEntity<Boolean>(isNewUser,HttpStatus.OK);
     }
 
+    @Override
+    public ResponseEntity<List<ChatMessageDto>> usersUserId1ConversationsUserId2Get(@ApiParam(value = "Numeric ID of the user1",required=true) @PathVariable("userId1") Long userId1, @ApiParam(value = "Numeric ID of the user2",required=true) @PathVariable("userId2") Long userId2) throws Exception {
+        if (!isRequesterAndAuthenticatedUserTheSame(userId1)){
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+        User receiver = userService.getUserById(userId2);
+        try {
+            List<ChatMessage> messages = chatService.getAllMessages(receiver);
```

Comment:
This is not an authentication logic, but authorization check. Since you mentioned this, it should be done in Service layer and I will move it there.

Reply:
Yes, sorry about mixing up the terms — I was thinking about the fact that the service layer accesses the current user, instead of receiving a userId from the controller. I was aiming to have only the controllers access the current user from the context, and then pass its ID to the lower layers.

That being said you are right, it's probably better to have this in the service layer.